src/System.Linq/tests/AverageTests.cs

```diff
@@ -39,6 +39,15 @@ public void EmptyNullableFloatSource()
         }
 
         [Fact]
+        public void EmptyNullableFloatSourceWithSelector()
```

Comment:
In some of the methods, like ```double? Min<TSource>(this IEnumerable<TSource> source, Func<TSource, double?> selector)```, there's an initial do...while loop, with the first line:
```C#
if (!e.MoveNext()) return value;
```
Would that be better as:
```
if (!e.MoveNext()) return null;
```
?